### ISSUE TITLE
Possible bug -channel-fx when writing to tiff

### ISSUE THREAD

**Issue by fmw42** [ImageMagick/ImageMagick#5843]
### ImageMagick version

7.1.0-52

### Operating system

MacOS

### Operating system, version and so on

Monterey

### Description

The following works fine to add an alpha channel to a PNG

Image:

![lena](https://user-images.githubusercontent.com/13458218/205368357-7ca29fc7-cb13-4ae2-a25d-d9fce3c1888d.png)

Gray Image:

![circle](https://user-images.githubusercontent.com/13458218/205368537-a2f74b41-e956-46ad-9a47-b825c79eec49.gif)

`magick lena.png circle.gif -channel-fx '| gray=>alpha' lena_circle.png`

![lena_circle](https://user-images.githubusercontent.com/13458218/205368637-e9dfd224-71a8-4241-b2fb-0be7a92b7bb1.png)

But when I try to do the same with TIFF output, it looks corrupt or strange.

`magick lena.png circle.gif -channel-fx '| gray=>alpha' lena_circle.tiff`

[lena_circle.tiff.zip](https://github.com/ImageMagick/ImageMagick/files/10143698/lena_circle.tiff.zip)

### Steps to Reproduce

See above

### Images

See above

**Comment by snibgo** [comment]
I confirm the problem with IM v7.1.0-49 on Windows.

A workaround is to insert "+channel" before writing the output. Windows syntax:
```
magick lena.png circle.gif -channel-fx "| gray=>alpha" +channel lena_circle.tiff
```
"+channel" should not be needed, of course.

**Comment by fmw42** [comment]
Yes, that works.  So I can either get background transparency by using -define tiff:alpha=associated or unassociated (not sure why both work) or I can get an alpha channel mask using -define tiff:alpha=unspecified

`magick lena.png \( circle.gif -negate \) -channel-fx "| gray=>alpha" +channel -define tiff:alpha=associated lena_circle_ass.tiff`

(but I also get two more totally black alpha channels)

`magick lena.png \( circle.gif -negate \) -channel-fx "| gray=>alpha" +channel -define tiff:alpha=unspecified lena_circle_unsp.tiff`

But I cannot seem to get both. Perhaps the define walks on the previous define?

**Comment by snibgo** [comment]
Writing to "ftxt:" provides some insight.

```
C:\web\im>%IMG7%magick xc:sRGB(10%,20%,30%) xc:gray(56%) -channel-fx "| gray=>alpha" ftxt:
0,0:36699.6
```
Note that this writes only one channel.
```
C:\web\im>%IMG7%magick xc:sRGB(10%,20%,30%) xc:gray(56%) -channel-fx "| gray=>alpha" +channel ftxt:
0,0:6553.5,13107,19660.5,36699.6,9.64288e-39,1.11122e-38
```
Note that this writes 6 channels. Six! The first four are RGBA, and the final two are spurious.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Comment by fmw42** [comment]
Yes, I noticed that there were two extra alpha channels when I opened the file in Photoshop.  See above

**Comment by fmw42** [comment]
Seems to be fixed now in 7.1.0.53
